src/Microsoft.Security.Utilities.Core/PreciselyClassifiedSecurityKeys/IdentifiableKey.cs

```diff
@@ -13,22 +13,22 @@ namespace Microsoft.Security.Utilities
 {
     public abstract class IdentifiableKey : RegexPattern, IIdentifiableKey, IHighPerformanceScannableKey
     {
-        public IdentifiableKey()
+        protected IdentifiableKey(string signature)
         {
             RotationPeriod = TimeSpan.FromDays(365 * 2);
             DetectionMetadata = DetectionMetadata.Identifiable;
+            Signatures = signature.ToSet();
+            ChecksumSeeds = Array.Empty<ulong>();
         }
 
         // Identifiable key patterns must have exactly one signature.
         internal string Signature => Signatures!.Single();
 
-        public string RegexNormalizedSignature => Regex.Escape(Signature);
```

Comment:
On a prior PR of mine, @rwoll asked for this to be renamed, but I left it since it was pre-existing. 

Looking at it again now, given that:
1. Its expense was non-obvious (main point of this PR)
2. We don't need to store it long-term
3. The contentious name is longer than the implementation

I've just inlined `Regex.Escape` when needed.